### PR TITLE
Lejit app : Error Handling and Firebase Authentication

### PR DESCRIPTION
**Pull request by GoutamShanbhag**
@yogendrakvr 

Lejit app : Message Modal is used to show confirmation or warning messages during sign up and sign in

### CODE REVIEW

File: React-Template-master/.gitignore [hunk 1]
```diff
@@ -20,6 +20,7 @@
 .env.development.local
 .env.test.local
 .env.production.local
+.env
```
Comment (yogendrakvr) [inline React-Template-master/.gitignore]: Why is .env added to gitignore?

File: React-Template-master/src/components/ErrorHandling.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export const getError = (error: AuthError): string => {
+    switch (error.code) {
+        case AuthErrorCodes.USER_DELETED:
+            return 'userNotFound';
+
+        case AuthErrorCodes.EMAIL_EXISTS:
+            return 'emailExists';
+
+        case AuthErrorCodes.UNVERIFIED_EMAIL:
+            return '';
```
Comment (yogendrakvr) [inline React-Template-master/src/components/ErrorHandling.ts]: Why does unverified email doesn't have a message?

File: React-Template-master/src/components/Firebase.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+const firebaseConfig = {
+    apiKey: process.env.REACT_APP_API_KEY,
+    authDomain: process.env.REACT_APP_AUTH_DOMAIN,
+    projectId: process.env.REACT_APP_PROJECT_ID,
+    storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
+    messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
+    appId: process.env.REACT_APP_APP_ID
+};
+
+// Initialize Firebase
```
Comment (yogendrakvr) [inline React-Template-master/src/components/Firebase.tsx]: I think there is no need for this comment here

File: React-Template-master/src/components/Logo.tsx [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
+import Correct from '../assets/Correct.svg';
+import Warning from '../assets/Warning.svg';
 // ----------------------------------------------------------------------
 
 interface LogoProps {
     sx?: SxProps;
 }
 
-export function Logo(sx: LogoProps): JSX.Element {
+export const Logo = (sx: LogoProps): JSX.Element => {
```
Comment (yogendrakvr) [inline React-Template-master/src/components/Logo.tsx]: Instead of replicating the same code again, can we add a prop to pass the icon to the logo component.

File: React-Template-master/src/components/MessageModal.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { Modal, Box, Typography, Button, useTheme } from '@mui/material';
+import { NEUTRAL } from '../theme/palette';
+import { CorrectLogo, WarningLogo } from './Logo';
+import { useTranslation } from 'react-i18next';
+
+interface ConfirmationModalProps {
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: The file name and component name doesn't match with the interface name. Make sure you give proper names

File: React-Template-master/src/components/MessageModal.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    open: boolean;
+    setOpen: Function;
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: Don't mention the type as Function, it varies depends on the function, give a more specific type

File: React-Template-master/src/components/MessageModal.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    modalContent: {
+        title: string;
+        subtitle: string;
+        type?: string;
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: Create an enum or type for the type of the model content, string is way too generalized

File: React-Template-master/src/components/MessageModal.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    transform: 'translate(-50%, -50%)',
+    width: '408px',
+    borderRadius: '4px',
+    height: '268px',
+    boxShadow: 24,
+    p: '20px',
+    boxSizeing: 'content-box'
+};
+
+export const MessageModal = (props: ConfirmationModalProps): JSX.Element => {
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: You can destructure the props here itself

File: React-Template-master/src/components/PasswordField.tsx [hunk 9]
```diff
@@ -59,6 +62,9 @@ export const PasswordField = ({
             type={passwordVisibility ? 'text' : 'password'}
             id="password"
             autoComplete="new-password"
+            helperText={
+                showHelperText && iconVisibility && password.length < 6 ? t('invalidPassword') : ''
```
Comment (yogendrakvr) [inline React-Template-master/src/components/PasswordField.tsx]:
```suggestion
                (showHelperText && iconVisibility && password.length < 6 )?? t('invalidPassword')
```

File: React-Template-master/src/pages/ForgotPassword.tsx [hunk 10]
```diff
@@ -1,18 +1,41 @@
...
+import React, { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import LoadingButton from '@mui/lab/LoadingButton';
 import { emailValidation } from '../components/EmailValidation';
+import { auth } from '../components/Firebase';
+import { AuthError, sendPasswordResetEmail } from 'firebase/auth';
+import { getError } from '../components/ErrorHandling';
+import { MessageModal } from '../components/MessageModal';
+
+const MODAL_CONTENT = {
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/ForgotPassword.tsx]: Why is the name in PascalCase, if you wanted an enum define an enum instead of object

**Review by yogendrakvr — CHANGES_REQUESTED** [review]

File: React-Template-master/src/components/Logo.tsx (hunk 4 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/Logo.tsx]: Okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 5 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/components/ErrorHandling.ts (hunk 2 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/ErrorHandling.ts]: okay

File: React-Template-master/src/components/Firebase.tsx (hunk 3 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/Firebase.tsx]: okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 7 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 8 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/components/PasswordField.tsx (hunk 9 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/PasswordField.tsx]: okay

File: React-Template-master/src/pages/ForgotPassword.tsx (hunk 10 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/ForgotPassword.tsx]: @yogendrakvr I have added a type but i didn't change the name style as MODAL_CONTENT is a constant

File: React-Template-master/src/components/MessageModal.tsx (hunk 6 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/components/Logo.tsx [hunk 11]
```diff
@@ -1,22 +1,35 @@
...
 import { Box, SxProps } from '@mui/material';
-import AppLogo from '../assets/Lejit.svg';
+
 // ----------------------------------------------------------------------
 
+export type SupportedLogoType = 'app' | 'modal';
+
 interface LogoProps {
     sx?: SxProps;
+    type: SupportedLogoType;
```
Comment (yogendrakvr) [inline React-Template-master/src/components/Logo.tsx]: Instead of type, It would be better to have a height and width prop.

File: React-Template-master/src/components/Logo.tsx [hunk 12]
```diff
@@ -1,22 +1,35 @@
...
+enum AppDimension {
+    width = '64px',
+    height = '33.6px'
 }
 
-export function Logo(sx: LogoProps): JSX.Element {
+enum ModalDimension {
+    width = '53.33px',
+    height = '53.33px',
+    top = '25.33px'
```
Comment (yogendrakvr) [inline React-Template-master/src/components/Logo.tsx]: What is the use of `top` property?

File: React-Template-master/src/components/MessageModal.tsx [hunk 13]
```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { Modal, Box, Typography, Button, useTheme } from '@mui/material';
+import Correct from '../assets/Correct.svg';
+import Warning from '../assets/Warning.svg';
+import { Logo } from './Logo';
+import { useTranslation } from 'react-i18next';
+
+export type SupportedModalType = 'correct' | 'warning';
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]:
```suggestion
export type SupportedModalType = 'success' | 'error';
```

File: React-Template-master/src/components/MessageModal.tsx [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+    const theme = useTheme();
+    const isCorrect = type === 'correct';
+    return (
+        <Box>
+            <Modal
+                open={open}
+                onClose={(): void => {
+                    setOpen(false);
+                }}>
+                <Box sx={{ ...style, bgcolor: theme.palette.common.white }}>
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: Make this component a styled component instead of setting style props

File: React-Template-master/src/components/MessageModal.tsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+    height: '268px',
+    boxShadow: 24,
+    p: '20px',
+    boxSizeing: 'content-box'
+};
+
+export const MessageModal = ({
+    open,
+    setOpen,
+    modalContent: { type, title, subtitle, buttonText }
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: There is no need to have this props inside an object, have them in the same level as open props.
Eg:
`interface MessageModalProps {
    open: boolean;
    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
    title: string;
    subtitle: string;
    type?: 'success'|'error';
    buttonText: string;
}`

File: React-Template-master/src/components/MessageModal.tsx [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+    const isCorrect = type === 'correct';
+    return (
+        <Box>
+            <Modal
+                open={open}
+                onClose={(): void => {
+                    setOpen(false);
+                }}>
+                <Box sx={{ ...style, bgcolor: theme.palette.common.white }}>
+                    {type &&
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: Create a function that return the relevant icon

File: React-Template-master/src/components/MessageModal.tsx [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+                        (isCorrect ? (
+                            <Logo type={'modal'} src={Correct} />
+                        ) : (
+                            <Logo type={'modal'} src={Warning} />
+                        ))}
+                    <Typography variant="subtitle1" sx={{ mt: '11.3px' }}>
+                        {t(title)}
+                    </Typography>
+                    <Typography
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: Create a styled component

File: React-Template-master/src/components/MessageModal.tsx [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+                            width: '360px',
+                            height: '48px'
+                        }}>
+                        {t(subtitle)}
+                    </Typography>
+                    <Button
+                        fullWidth
+                        sx={{
+                            mt: '24px',
+                            position: 'absolute',
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: Why is this needed?

File: React-Template-master/src/components/MessageModal.tsx [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+                            bottom: '20px',
+                            width: '360px'
+                        }}
+                        variant={isCorrect ? 'contained' : 'outlined'}
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx:85]: What is button variant when the type is undefned

File: React-Template-master/src/components/PasswordField.tsx [hunk 20]
```diff
@@ -59,6 +62,9 @@ export const PasswordField = ({
             type={passwordVisibility ? 'text' : 'password'}
             id="password"
             autoComplete="new-password"
+            helperText={
```
Comment (yogendrakvr) [inline React-Template-master/src/components/PasswordField.tsx:65]: Will the helper text always be invalid password?

Comment (yogendrakvr) [reply React-Template-master/src/components/PasswordField.tsx:65]: What if you want to show a different message

File: React-Template-master/src/components/MessageModal.tsx [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+                }}>
+                <Box sx={{ ...style, bgcolor: theme.palette.common.white }}>
+                    {type &&
+                        (isCorrect ? (
+                            <Logo type={'modal'} src={Correct} />
+                        ) : (
+                            <Logo type={'modal'} src={Warning} />
+                        ))}
+                    <Typography variant="subtitle1" sx={{ mt: '11.3px' }}>
+                        {t(title)}
```
Comment (yogendrakvr) [inline React-Template-master/src/components/MessageModal.tsx]: Instead of sending the key it would better to send the translated text to the component.
In that case the parent will decide what the text is shown instead of the child component deciding what the text will be.

File: React-Template-master/src/pages/Register.tsx [hunk 22]
```diff
@@ -29,16 +42,42 @@ export const Register = (): JSX.Element => {
...
-    const [isValidEmail, setIsValidEmail] = useState(true);
-
     const handleSubmit = async (event: React.FormEvent): Promise<void> => {
         event.preventDefault();
-        //TODO: Register the user and add all the data into firestore
+        setLoading(true);
+        if (data.email && data.password) {
+            const { firstName, lastName, email, password } = data;
+            try {
+                const newUser = await createUserWithEmailAndPassword(auth, email, password);
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/Register.tsx]: create a separate file inside firebase to add auth and database functions and just call the functions here

File: React-Template-master/src/pages/Register.tsx [hunk 23]
```diff
@@ -29,16 +42,42 @@ export const Register = (): JSX.Element => {
         email: '',
         password: ''
     });
+    const [open, setOpen] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const [{ isError, errorMessage }, setHelperText] = useState({
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/Register.tsx]: There is no need to have both isError and errorMessage, we can only use the errorMessage to identify if error is present or not

Comment (yogendrakvr) [inline React-Template-master/src/pages/Register.tsx]:
```suggestion
    const [error, setError] = useState('')
```

File: React-Template-master/src/pages/Register.tsx [hunk 24]
```diff
@@ -84,28 +124,33 @@ export const Register = (): JSX.Element => {
                     <Grid item xs={12}>
                         <TextField
                             value={data.email}
-                            error={!isValidEmail}
+                            error={isError}
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/Register.tsx]:
```suggestion
                            error={Boolean(error)}
```

File: React-Template-master/src/pages/Register.tsx [hunk 25]
```diff
@@ -84,28 +124,33 @@ export const Register = (): JSX.Element => {
...
                             onChange={async (e): Promise<void> => {
                                 setData({ ...data, email: e.target.value });
-                                setIsValidEmail(await emailValidation(e.target.value));
+                                setHelperText({
+                                    isError: !(await emailValidation(e.target.value)),
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/Register.tsx]: can you get the result of the email validation before updating the state,

File: React-Template-master/src/pages/SignIn.tsx [hunk 26]
```diff
@@ -18,23 +24,58 @@ const checkForEmptyInputs = (data: SignInFormValues): boolean => {
...
-    const handleSubmit = (event: React.FormEvent): void => {
+    const handleSubmit = async (event: React.FormEvent): Promise<void> => {
         event.preventDefault();
-        //TODO: Add firebase function to sign in and direct the page to dashboard
-        return;
+        setLoading(true);
+        if (data.email && data.password) {
+            const { email, password } = data;
+            try {
+                const user = await signInWithEmailAndPassword(auth, email, password);
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/SignIn.tsx]: create a separate file inside firebase to add auth and database functions and just call the functions here

File: React-Template-master/src/pages/SignIn.tsx [hunk 27]
```diff
@@ -56,14 +97,17 @@ export const SignIn = (): JSX.Element => {
                             value={data.email}
                             onChange={async (e): Promise<void> => {
                                 setData({ ...data, email: e.target.value });
-                                setIsValidEmail(await emailValidation(e.target.value));
+                                setHelperText({
+                                    isError: !(await emailValidation(e.target.value)),
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/SignIn.tsx]: please make the changes in other places based of the feecback.

**Review by yogendrakvr — CHANGES_REQUESTED** [review]

File: React-Template-master/.gitignore (hunk 1 above)
Comment (GoutamShanbhag) [reply React-Template-master/.gitignore]: okay

File: React-Template-master/src/components/Logo.tsx (hunk 11 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/Logo.tsx]: Okay

File: React-Template-master/src/components/Logo.tsx (hunk 12 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/Logo.tsx]: this is to position the icon from the top

File: React-Template-master/src/components/MessageModal.tsx (hunk 13 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 14 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: Okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 15 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: Okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 16 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 17 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/components/MessageModal.tsx (hunk 18 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: to position button from the bottom of the modal

File: React-Template-master/src/components/MessageModal.tsx (hunk 19 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx:85]: outlined

File: React-Template-master/src/components/PasswordField.tsx (hunk 20 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/PasswordField.tsx:65]: I have changed this part the code in my context branch. so that the  error message is not restricted to only invalid password.

To show different message we can pass the message in showHelperText prop

File: React-Template-master/src/components/MessageModal.tsx (hunk 21 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/MessageModal.tsx]: okay

File: React-Template-master/src/pages/Register.tsx (hunk 22 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/Register.tsx]: okay

File: React-Template-master/src/pages/Register.tsx (hunk 23 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/Register.tsx]: okay

File: React-Template-master/src/pages/Register.tsx (hunk 24 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/Register.tsx]: okay

File: React-Template-master/src/pages/Register.tsx (hunk 25 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/Register.tsx]: okay

File: React-Template-master/src/pages/SignIn.tsx (hunk 27 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/SignIn.tsx]: okay

File: React-Template-master/src/pages/SignIn.tsx (hunk 26 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/SignIn.tsx]: okay

File: React-Template-master/src/components/ChangePasswordModal.tsx [hunk 28]
```diff
@@ -1,37 +1,39 @@
-import React, { useState } from 'react';
-import { Modal, Grid, Box, TextField, useTheme, Typography, Button } from '@mui/material';
+import React from 'react';
+import { Modal, Grid, Box, TextField, Typography, Button, styled } from '@mui/material';
 import { NEUTRAL } from '../theme/palette';
 import { useTranslation } from 'react-i18next';
 
-const style = {
+//-------------------------------------------------
```
Comment (yogendrakvr) [inline React-Template-master/src/components/ChangePasswordModal.tsx]: Remove this

File: React-Template-master/src/components/ChangePasswordModal.tsx [hunk 29]
```diff
@@ -1,37 +1,39 @@
...
+const BoxStyle = styled(Box)(({ theme }) => ({
```
Comment (yogendrakvr) [inline React-Template-master/src/components/ChangePasswordModal.tsx]:
```suggestion
const BoxContainer = styled(Box)(({ theme }) => ({
```

File: React-Template-master/src/components/Firebase.tsx [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+interface UserSignIn {
+    email: string;
+    password: string;
+}
+
+const app = initializeApp(firebaseConfig);
+export const auth = getAuth(app);
+export const db = getFirestore(app);
+
+export const createNewUser = async (user: User): Promise<void> => {
```
Comment (yogendrakvr) [inline React-Template-master/src/components/Firebase.tsx]: Create a separate files for auth functions

File: React-Template-master/src/components/Firebase.tsx [hunk 31]
```diff
@@ -0,0 +1,53 @@
...
+    await setDoc(doc(db, `users`, newUser.user.uid), {
+        firstName,
+        lastName,
+        email
+    });
+
+    await sendEmailVerification(newUser.user);
+};
+
+export const logIn = async ({ email, password }: UserSignIn): Promise<UserCredential> => {
```
Comment (yogendrakvr) [inline React-Template-master/src/components/Firebase.tsx]:
```suggestion
export const logIn = async (email: string, password: string): Promise<UserCredential> => {
```

File: React-Template-master/src/components/Firebase.tsx (hunk 30 above)
Comment (yogendrakvr) [reply React-Template-master/src/components/Firebase.tsx]: Get the data as parameters instead of object

File: React-Template-master/src/pages/SignIn.tsx [hunk 32]
```diff
@@ -56,14 +84,18 @@ export const SignIn = (): JSX.Element => {
                             value={data.email}
                             onChange={async (e): Promise<void> => {
                                 setData({ ...data, email: e.target.value });
-                                setIsValidEmail(await emailValidation(e.target.value));
+                                if (!(await emailValidation(e.target.value))) {
```
Comment (yogendrakvr) [inline React-Template-master/src/pages/SignIn.tsx]: Store the await result in a variable before checking.

**Review by yogendrakvr — CHANGES_REQUESTED** [review]

File: React-Template-master/src/components/ChangePasswordModal.tsx (hunk 28 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/ChangePasswordModal.tsx]: Okay

File: React-Template-master/src/components/ChangePasswordModal.tsx (hunk 29 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/ChangePasswordModal.tsx]: okay

File: React-Template-master/src/components/Firebase.tsx (hunk 30 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/Firebase.tsx]: Okay

File: React-Template-master/src/components/Firebase.tsx (hunk 31 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/Firebase.tsx]: okay

File: React-Template-master/src/pages/SignIn.tsx (hunk 32 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/pages/SignIn.tsx]: okay

File: React-Template-master/src/components/Firebase.tsx [hunk 33]
```diff
@@ -0,0 +1,53 @@
+import { initializeApp } from 'firebase/app';
```
Comment (yogendrakvr) [inline React-Template-master/src/components/Firebase.tsx]: Remove this file

**Review by yogendrakvr — CHANGES_REQUESTED** [review]

File: React-Template-master/src/components/Firebase.tsx (hunk 33 above)
Comment (GoutamShanbhag) [reply React-Template-master/src/components/Firebase.tsx]: okay

**Review by yogendrakvr — APPROVED** [review]
